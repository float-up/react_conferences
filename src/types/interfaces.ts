export interface IProgramScope {
    level: 'HOT' | 'INTERMEDIATE' | 'ADVANCED' | 'HARDCORE' | 'ACADEMIC';
    lang: 'RU' | 'EN';
}

export interface IProgram {
    id: string;
    title: string;
    lector: string;
    type: IProgramScope;
}

