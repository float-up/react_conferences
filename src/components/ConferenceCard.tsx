import React from "react";
import {IProgram} from "../types/interfaces";

interface IProps {
    confInfo: IProgram;
}

export const ConferenceCard: React.FC<IProps> = props => {

    const {
        confInfo
    } = props;

    return (
        <div className="card grey lighten-5">
            <div className="card-content black-text">
                <span className="card-title">{confInfo.title}</span>
                <br/>
                <p className="card-lector">{confInfo.lector}</p>
                <br/>
                <p className="card-level">{confInfo.type.level} / {confInfo.type.lang}</p>
            </div>
        </div>
    )
}

