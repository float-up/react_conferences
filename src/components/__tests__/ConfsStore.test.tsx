import React from "react";
import {shallow} from "enzyme";
import {ConferenceCard} from "../ConferenceCard";
import {IProgram} from "../../types/interfaces";

const info: IProgram = {
    "id": "1",
    "title": "Client rendering, server rendering, pre rendering. The full spectrum of JS website and application perfomance delivery",
    "lector": "Guillermo Rauch",
    "type": {
        "level": "INTERMEDIATE",
        "lang": "EN"
    }
}

let wrapper = shallow(<ConferenceCard confInfo={info}/>)
describe('ConferenceCard', () => {
    it('test ConferenceCard render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('test show title into ConferenceCard', () => {
        expect(wrapper.find('span').text()).toEqual('Client rendering, server rendering, pre rendering. The full spectrum of JS website and application perfomance delivery')
    })

    it('test show lector into ConferenceCard', () => {
        expect(wrapper.find('p.card-lector').text()).toEqual('Guillermo Rauch')
    })

    it('test show lector into ConferenceCard', () => {
        expect(wrapper.find('p.card-level').text()).toEqual('INTERMEDIATE / EN')
    })
})
