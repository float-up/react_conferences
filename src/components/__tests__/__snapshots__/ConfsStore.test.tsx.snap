// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConferenceCard test ConferenceCard render 1`] = `
<div
  className="card grey lighten-5"
>
  <div
    className="card-content black-text"
  >
    <span
      className="card-title"
    >
      Client rendering, server rendering, pre rendering. The full spectrum of JS website and application perfomance delivery
    </span>
    <br />
    <p
      className="card-lector"
    >
      Guillermo Rauch
    </p>
    <br />
    <p
      className="card-level"
    >
      INTERMEDIATE
       / 
      EN
    </p>
  </div>
</div>
`;
